django/core/cache/backends/locmem.py

```diff
@@ -13,117 +13,97 @@
 _locks = {}
 
 
-@contextmanager
-def dummy():
-    """A context manager that does nothing special."""
-    yield
-
-
 class LocMemCache(BaseCache):
     def __init__(self, name, params):
         super().__init__(params)
-        self._cache = _caches.setdefault(name, {})
+        self._cache = _caches.setdefault(name, OrderedDict())
         self._expire_info = _expire_info.setdefault(name, {})
-        self._lock = _locks.setdefault(name, RWLock())
+        self._lock = _locks.setdefault(name, RLock())
 
     def add(self, key, value, timeout=DEFAULT_TIMEOUT, version=None):
         key = self.make_key(key, version=version)
         self.validate_key(key)
         pickled = pickle.dumps(value, pickle.HIGHEST_PROTOCOL)
-        with self._lock.writer():
+        with self._lock:
             if self._has_expired(key):
                 self._set(key, pickled, timeout)
                 return True
             return False
 
-    def get(self, key, default=None, version=None, acquire_lock=True):
+    def get(self, key, default=None, version=None):
```

Comment:
I am not sure we can just drop an argument, it might need to go through the deprecation timeline. Whilst added to support an internal function, `incr()` being threadsafe in 6448dd833524ac3fc503506b624841c9d642de8a, there's a potential other code out there is using it. Thoughts @jarshwah ?

Reply:
FWIW, the dropped argument is undocumented. I was surprised to see it there.

Reply:
I've done a code search to look for open source uses, but can't find anything. A lot of false positives, looks like `acquire_lock` is a generally popular method name: https://github.com/search?utf8=%E2%9C%93&q=get+acquire_lock+django+-filename%3Alocmem.py+-filename%3Asynch.py+language%3APython&type=Code .

I agree, it was undocumented and not part of the general cache API. Still the deprecation path isn't that bad, it just involves a dummy argument that throws a warning until Django 3.0.

Reply:
What's the most idiomatic way to write that? I came up with:

```python
    def get(self, key, default=None, version=None, **kwargs):
        if 'acquire_lock' in kwargs:
            warnings.warn(
                'acquire_lock argument is unnecessary and deprecated'
                RemovedInDjango30Warning,
            )
```

But that assumes ``acquire_lock`` was given as a keyword argument.

Reply:
I think maybe

```py
dummy = object()

def get(self, key, default=None, version=None, acquire_lock=dummy):
    if acquire_lock is not dummy:
         warnings.warn(
```

I still think this arg fits in the grey area for deprecation timeline, would like someone to weigh in e.g. @jarshwah

Reply:
`acquire_lock` was added for internal usage in 6448dd833524ac3fc503506b624841c9d642de8a, so I don't see a need for a deprecation.

Reply:
Ok thanks Tim